spec/component/models/duplicate_publication_group_spec.rb

```diff
@@ -526,11 +526,54 @@
       it 'groups the publications' do
         described_class.group_duplicates_of(p1)
         group = p1.reload.duplicate_group
+        expect(p2.reload.duplicate_group).to eq group
+      end
+    end
+
+    context 'given a publication with a similar title to another publication and a different publication date but the year is within +/-2' do
+      let!(:p1) { create :publication,
+                         title: 'A Typical Title',
+                         published_on: Date.new(2000, 5, 20),
+                         doi: nil }
+
+      let!(:p2) { create :publication,
+                         title: 'The Typical Title',
+                         published_on: Date.new(2002, 1, 1),
+                         doi: nil }
+      let!(:p3) { create :publication,
+                         title: 'The Typical Title 3',
+                         published_on: Date.new(2000, 5, 20),
+                         doi: nil }
 
+      it 'groups the publications' do
+        described_class.group_duplicates_of(p1)
+        group = p1.reload.duplicate_group
```

Comment:
I'm tempted to have a 'not be nil' check in here, but I can go either way.  This test works assuming p1 and p3 end up grouping too, which isn't guaranteed if we are making changes to the grouping logic and something goes wrong.  It may be unlikely, but we could still end up with a nil == nil situation.